lib/score_card.js

```diff
@@ -0,0 +1,119 @@
+function ScoreCard(){
+  this.total = 0
+  this.frame = new Frame()
+  this.frames = []
+  this.frameCount = 1
+}
+
+ScoreCard.prototype.recordRoll = function(points){
+  this.frameCheck()
+  if(this.cheatGuard(points)==true){
+    return 'illegal roll'
+  } else {
```

Comment:
Two things - firstly, we don't really need the 'else' and secondly we should encapsulate the validation logic within the method. I would expect to see something like:

```
if (this.isLegalRoll(points)) {
  return 'Illegal Roll';
}

... code here ... 
``` 
Or even better

```
assertIsLegalRoll(points);

... code here ... 
``` 

Where assertLegalRoll throws an exception if it is not legal, and the calling code is responsible for deciding what to do with that exception (because asserting a rule and handling the result of that assertion are two different responsibilities)